src/anomalib/engine/engine.py

```diff
@@ -403,7 +405,22 @@ def predict(
         """
         if model:
             self._setup_trainer(model)
-            self._setup_dataset_task(dataloaders, datamodule)
+
+        if not ckpt_path:
+            logger.warning("ckpt_path is not provided. Model weights will not be loaded.")
+
+        if datamodule is not None and isinstance(datamodule, Dataset):
+            dataloader = DataLoader(datamodule)
+            datamodule = None
+            if dataloaders is None:
+                dataloaders = dataloader
+            elif isinstance(dataloaders, DataLoader):
+                dataloaders = [dataloaders, dataloader]
+            else:  # dataloader is a list
+                dataloaders = [*dataloaders, dataloader]
```

Comment:
I've recently created this #1576. Would this part resolve that issue?

Reply:
No, this only handles the instance when a dataset is passed to the predict method. I've added a comment now to inform this.